src/compiler/optim.jl

```diff
@@ -173,13 +228,13 @@ function wrap_entry!(ctx::CompilerContext, mod::LLVM.Module, entry_f::LLVM.Funct
     linkage!(entry_f, LLVM.API.LLVMInternalLinkage)
 
     # set calling convention to AMDGPU_KERNEL
-    # NOTE: only valid to set this on the outermost function
+    # NOTE: only valid to set this on the outermost function -- @julian: special add??
```

Comment:
Yep, this is required by the backend to indicate that the outermost function is a kernel function, for whatever internal purposes they require that for. We can leave this as-is.